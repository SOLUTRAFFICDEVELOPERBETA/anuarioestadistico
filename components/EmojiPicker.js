/* eslint-disable react/display-name */
import React from 'react';
import PropTypes from 'prop-types';
import usePopover from '../hooks/usePopover';
import { Button } from '@material-ui/core';
import { InsertEmoticonTwoTone } from '@material-ui/icons';
import dynamic from 'next/dynamic';
import Spinner from './Spinner';

const Picker = dynamic(import('emoji-picker-react'), {
    ssr: false,
    loading: () => <Spinner />
});

/**
 * @description Componente que permite Visualizar Emojis y seleccionar en cada texto.
 * @param {func: () => void} onSelect
 * @param {string} id
 * Consulte los elementos utilizados en {@link https://material-ui.com/ | Material-ui}
 */
const EmojiPicker = ({ onSelect, id }) => {
    const [handleOpen, PopoverComponent] = usePopover();
    const handleEmojiClick = (event, emojiObject) => onSelect(emojiObject);
    

    return (
        <React.Fragment>
            <Button color="primary" size="small" onClick={handleOpen}>
                <InsertEmoticonTwoTone />
            </Button>
            <PopoverComponent id={`popover-emoji-${id}`}>
                <Picker onEmojiClick={handleEmojiClick} />
            </PopoverComponent>
        </React.Fragment>
    );
};

EmojiPicker.propTypes = {
    onSelect: PropTypes.func.isRequired,
    id: PropTypes.string.isRequired
};

export default EmojiPicker;
