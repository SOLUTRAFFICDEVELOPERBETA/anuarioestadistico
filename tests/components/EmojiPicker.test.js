import '@testing-library/jest-dom';

import { shallow, } from 'enzyme';
import React from 'react';
import EmojiPicker from '../../components/EmojiPicker';

describe('Pruebas en <EmojiPicker/>', () => {
    const onSelect = jest.fn(() => {})
    const wrapper = shallow(
        <EmojiPicker
            id={'idEmojiPiker'}
            onSelect={onSelect}

        />
    )

    test('debe mostrarse correctamente testing pendiende ', () => {
        expect(wrapper).toMatchSnapshot()
    });


    test('debe llamar la funciÃ³n onSelect()', () => {
        wrapper.find('ForwardRef(LoadableComponent)').prop('onEmojiClick')({ emojiObject: expect.any });

        expect(onSelect).toHaveBeenCalled();
        expect(onSelect).toHaveBeenCalledTimes(1)
    })
})