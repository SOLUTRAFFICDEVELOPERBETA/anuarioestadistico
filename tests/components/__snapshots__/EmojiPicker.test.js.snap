// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <EmojiPicker/> debe mostrarse correctamente testing pendiende  1`] = `
<Fragment>
  <WithStyles(ForwardRef(Button))
    color="primary"
    onClick={[Function]}
    size="small"
  >
    <InsertEmoticonTwoToneIcon />
  </WithStyles(ForwardRef(Button))>
  <PopoverComponent
    id="popover-emoji-idEmojiPiker"
  >
    <ForwardRef(LoadableComponent)
      onEmojiClick={[Function]}
    />
  </PopoverComponent>
</Fragment>
`;
